### PR TITLE
[O11y][Azure] Migration of Azure AD Identity Protection dashboard to Lens

### LINKED ISSUE #7583: [O11y][Azure] Incorrect filter in Azure AD Identity Protection dashboard
## Description
The integration using `azure.identityprotection` for field names and `azure.identity_protection` as dataset.  Hence, need to replace `azure.identityprotection` with `azure.identity_protection` as the dataset used for queries in the dashboard.

### Screenshot
image

### PR DESCRIPTION
**Pull request by rajvi-elastic**
### Urgency
- High

### Activity Type
- Enhancement

## What does this PR do?
- Manually migrate `Azure AD Identity Protection` visualizations to the lens in the current Kibana version `8.6.0` itself. 

- Statistics for `Azure AD Identity Protection` Lens migration:
## Migration stats
<html>
<body>
<google-sheets-html-origin>

  | Before Migration |  | After Migration |  | 
-- | -- | -- | -- | -- 
Dashboad Name | Lens | Visualization | Lens | Visualization
[Logs Azure] Azure AD Identity Protection | 3 | 2 | 5 | 0

</body>
</html>

## Checklist

- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that panels are populated with data.
- [X] I have verified that panels are not distorted after being migrated to the lens.
- [x] I have updated screenshots of the dashboard.
- [X] I have verified that the data counts are matching and panel aggregations are the same as before.

## Author's Checklist

- [X] Migrated panels should be removed from visualization folder.
- [X] Migrated visualizations are populating in current Kibana version 8.6.0 itself.

## Related issues

- Relates #7230 
- Closes #7583

## Screenshot
Note: To get actual data is not straightforward in this case. User have to trigger it accessing from an unknown ip in order to generate these logs, and also it's a premium feature. Hence dashboard is not populated.

**Dashboard is getting populated after lens migration and updating data-stream filter.**
image

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Comment by zmoog** [comment]
Hey @rajvi-elastic, I know setting up a real Identity Protection is not straightforward and involves enabling an Active Directory Premium P2 license.

However, for the purpose of this PR, we don't have to set up the whole thing; we can send sample event logs to an event hub using a CLI tool:

```text
┌─────────────┐         ┌───────────┐        ┌───────────────┐
│  event hub  │────────▶│   agent   │───────▶│  data stream  │
└─────────────┘         └───────────┘        └───────────────┘
       ▲                                                      
                                                              
       │                                                      
     send                                                     
    sample                                                    
    events                                                    
       │                                                      
┌─────────────┐                                               
│  cli tool   │                                               
└─────────────┘                                               
```

**Comment by zmoog** [comment]
We have a few [sample events](https://github.com/elastic/integrations/tree/main/packages/azure/data_stream/identity_protection/_dev/test/pipeline) in the repo, and I probably have more on my laptop; we can use these as a template. 

To send data, we'll soon have support for the event hub in https://github.com/elastic/stream, but in the meantime, you can use https://github.com/zmoog/eventhubs to read sample event logs and publish them in the event hub.

Let me know if you need more information. I can put together a short guide if you need it.

**Review by zmoog — CHANGES_REQUESTED** [review]
We should test the dashboard sending sample events using the CLI tool.

**Comment by zmoog** [comment]
The problem with this integration is that it's using `azure.identityprotection` for field names and `azure.identity_protection` as dataset. We usually use the same for both. This wasn't a wise choice, and it's on me.

We need to have just one of the two eventually, but the very first change we need to to is to make the dashboard work.

File: packages/azure/kibana/dashboard/azure-5ee36c30-32dc-11ed-a2e6-916b60bbea71.json [hunk 1]
```diff
@@ -1,748 +1,809 @@
...
-                        "bd75d029-b44e-4c22-b06e-df0e4da1e694": {
-                          "dataType": "number",
-                          "isBucketed": false,
-                          "label": "Count of records",
-                          "operationType": "count",
-                          "scale": "ratio",
-                          "sourceField": "___records___"
+                        "query": {
+                            "match_phrase": {
+                                "data_stream.dataset": "azure.identityprotection"
```
Comment (zmoog) [inline packages/azure/kibana/dashboard/azure-5ee36c30-32dc-11ed-a2e6-916b60bbea71.json:24]: Update dataset to fix the dashboard:

```suggestion
                                "data_stream.dataset": "azure.identityprotection"
```

File: packages/azure/kibana/dashboard/azure-5ee36c30-32dc-11ed-a2e6-916b60bbea71.json [hunk 2]
```diff
@@ -1,748 +1,809 @@
...
+                            "store": "appState"
+                        },
+                        "meta": {
+                            "alias": null,
+                            "disabled": false,
+                            "indexRefName": "kibanaSavedObjectMeta.searchSourceJSON.filter[0].meta.index",
+                            "key": "data_stream.dataset",
+                            "negate": false,
+                            "params": {
+                                "query": "azure.identityprotection"
```
Comment (zmoog) [inline packages/azure/kibana/dashboard/azure-5ee36c30-32dc-11ed-a2e6-916b60bbea71.json:18]: Update dataset to fix the dashboard:

```suggestion
                                "query": "azure.identity_protection"
```

**Review by zmoog — CHANGES_REQUESTED** [review]
@rajvi-elastic, I added a couple of suggestions to replac `azure.identityprotection` with `azure.identity_protection` as the dataset used for queries in the dashboard.

This makes the dashboard display data.

**Comment by zmoog** [comment]
The priority is to make the integration work; after it's merged, I'll resolve the naming inconsistencies.

File: packages/azure/kibana/dashboard/azure-5ee36c30-32dc-11ed-a2e6-916b60bbea71.json (hunk 1 above)
Comment (rajvi-elastic) [reply packages/azure/kibana/dashboard/azure-5ee36c30-32dc-11ed-a2e6-916b60bbea71.json:24]: Here also shouldn't it be `"data_stream.dataset": "azure.identity_protection"`?

Comment (zmoog) [reply packages/azure/kibana/dashboard/azure-5ee36c30-32dc-11ed-a2e6-916b60bbea71.json:24]: Yep, definitely! I probably mis-pressed the underscore key! 🤦  Not it is fixed as `azure.identity_protection`.

**Comment by rajvi-elastic** [comment]
@zmoog, I have updated the filters as per the suggestion and dashboards are also getting populated. I have added screenshot in description.

**Review by zmoog — APPROVED** [review]
LGTM, thanks!

**Pull request merged**